src/api/interappbus/channel/index.ts

```diff
@@ -36,67 +37,65 @@ export class Channel extends EmitterBase {
             .then(({ payload }) => payload.data);
     }
 
-    public async onChannelConnect(listener: Function): Promise<void> {
-        await this.on('channel-connected', (payload) => {
-            const eventPayload = Object.assign(payload, {type: 'connected'});
-            listener(eventPayload);
-        });
+    public async onChannelConnect(listener: (...args: any[]) => void): Promise<void> {
+        await this.on('connected', listener);
     }
 
-    public async onChannelDisconnect(listener: Function): Promise<void> {
-        await this.on('channel-disconnected', (payload) => {
-            const eventPayload = Object.assign(payload, {type: 'disconnected'});
-            listener(eventPayload);
-        });
+    public async onChannelDisconnect(listener: (...args: any[]) => void): Promise<void> {
+        await this.on('disconnected', listener);
     }
 
-    // DOCS - if want to send payload, put payload in options
-    public async connect(options: Options): Promise<ChannelClient> {
-        const { uuid, name } = options;
-        let resolver: any;
-        let listener: any;
-        //@ts-ignore
-        // tslint:disable-next-line
+    public async connect(channelName: string, options?: Options): Promise<ChannelClient> {
```

Comment:
If options is optional, repeating channelName shouldn't be on the interface. Also is uuid still necessary?